File: code/mathematical-algorithms/binomial_coefficient/binomial_coefficient.c
// A Dynamic Programming based solution that uses table C[][] to
// calculate the Binomial Coefficient
#include<stdio.h>
 
// Prototype of a utility function that returns minimum of two integers
int min(int a, int b);
 
// Returns value of Binomial Coefficient C(n, k)
int binomialCoeff(int n, int k)
{
    int C[n+1][k+1];
    int i, j;
 
    // Caculate value of Binomial Coefficient in bottom up manner
    for (i = 0; i <= n; i++)
    {
        for (j = 0; j <= min(i, k); j++)
        {
            // Base Cases
            if (j == 0 || j == i)
                C[i][j] = 1;
 
            // Calculate value using previosly stored values
            else
                C[i][j] = C[i-1][j-1] + C[i-1][j];
        }
    }
 
    return C[n][k];
}
 
// A utility function to return minimum of two integers
int min(int a, int b)
{
    return (a<b)? a: b;
}
 
/* Drier program to test above function*/
int main()
{
    int n = 5, k = 2;
    printf ("Value of C(%d, %d) is %d ", n, k, binomialCoeff(n, k) );
    return 0;
}

File: code/dynamic_programming/src/binomial_coefficient/binomial_coefficient.cs
// C# Code for Dynamic Programming | 
using System; 

class GFG { 
	
	// Returns value of Binomial 
	// Coefficient C(n, k) 
	static int binomialCoeff(int n, int k) 
	{ 
		
		// Base Cases 
		if (k == 0 || k == n) 
			return 1; 
		
		// Recur 
		return binomialCoeff(n - 1, k - 1) + 
					binomialCoeff(n - 1, k); 
	} 
	
	/* Driver program to test above function */
	public static void Main() 
	{ 
		int n = 5, k = 2; 
		Console.Write("Value of C(" + n + ","
							+ k + ") is " + 
						binomialCoeff(n, k)); 
	} 
} 
 
